faq.md

```diff
@@ -0,0 +1,31 @@
+## FAQ questions
+
+### Why use a new ABI? Why not an attribute?
+- Unlike ABIs, there is no syntax for assigning attributes to function
+  pointers. The ABI must be part of the type system so that callers of function
+  pointers know whether or not the function may unwind.
+### What should I do if I want to have C++ functions that unwind into Rust? [DRAFT; TODO]
+- mark declarations in Rust `extern "C unwind"`
+- Do *not* use `catch_unwind` in calling code
+- compile w/ `panic = unwind`
+- Check compiler, linker, & platform documentation (TODO: expand on this)
+- Re-test for every compiler update
+- (TODO: other caveats?)
+### How does cross-language unwinding differ from cross-language
+  `setjmp`/`longjmp`?
+- `setjmp`/`longjmp` across Rust frames is currently guaranteed not to have
+  undefined behavior or to `abort`
+- It should never be assumed that `drop` will be called for objects in
+  intermediate frames traversed by a `longjmp`, but this may occur on certain
```

Comment:
If replacing a `longjmp` with a `panic` and a `setjmp` with a `catch_unwind` causes any destructors to run, the behavior is undefined. Some safe Rust abstractions like `Pin` rely on such things never happening.

Reply:
Well, so much for my "safe" phrasing above! But this is sort of what I was worried about and why I don't want to just say that the behavior is defined. Wasn't it pointed out in one of the RFC threads that LLVM with the MSVC linker currently _does_ generate code that causes `longjmp` to invoke destructors?

Reply:
> Wasn't it pointed out in one of the RFC threads that LLVM with the MSVC linker currently does generate code that causes longjmp to invoke destructors?

Since for the case being discussed the behavior is undefined, running destructors is one of the many things that could happen. Whatever LLVM does can change any time as a consequence of this, and IIRC whether destructors are run or not for particular functions depends on things like the optimization level.